src/otx/v2/adapters/torch/lightning/modules/datasets/dataset.py

```diff
@@ -0,0 +1,370 @@
+"""Dataset & DataModule for OTX lightning modules."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+from pytorch_lightning import LightningDataModule
+from torch.utils.data import DataLoader, Dataset
+from torchvision import transforms
+
+from otx.v2.adapters.torch.modules.utils.io import get_image_filenames, read_image
+from otx.v2.adapters.torch.modules.utils.mask_to_bbox import convert_polygon_to_mask, generate_bbox_from_mask
+from otx.v2.api.entities.dataset_item import DatasetItemEntity
+from otx.v2.api.entities.datasets import DatasetEntity
+from otx.v2.api.entities.image import Image
+from otx.v2.api.entities.label import LabelEntity
+from otx.v2.api.entities.scored_label import ScoredLabel
+from otx.v2.api.entities.shapes.polygon import Polygon
+from otx.v2.api.entities.subset import Subset
+from otx.v2.api.utils.logger import get_logger
+
+from .pipelines import MultipleInputsCompose, Pad, ResizeLongestSide, collate_fn
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    import albumentations as al
+    from omegaconf import DictConfig, ListConfig
+
+logger = get_logger()
+
+
+def get_transform(
```

Comment:
Should this be a private method? Do we refer to this anywhere else?

Reply:
Currently, this is the method used by `OTXVisualPromptingDataset` and `VisualPromptInferenceDataset`, so i'll leave it as is for now. It should be refactored together when these datasets are modified in the future.